src/components/patterns/on-the-page.tsx

```diff
@@ -0,0 +1,136 @@
+"use client"
+
+import React, {useState, useEffect} from "react"
+import {NodeStanfordNews, NodeStanfordPage, NodeSulLibrary} from "@/lib/gql/__generated__/drupal.d"
+import {twMerge} from "tailwind-merge"
+import DrupalLink from "./elements/drupal-link"
+import SelectList from "./elements/select-list"
+import {SelectOptionDefinition} from "@mui/base/useSelect"
+
+interface Heading {
+  text: string
+  id: string
+}
+
+const OnThePageLink = ({node}: {node: NodeStanfordPage | NodeStanfordNews | NodeSulLibrary}) => {
+  const [headings, setHeadings] = useState<Heading[]>([])
+  const [activeHeading, setActiveHeading] = useState<string>("")
+
+  const relLinkHeading = "sulRelLinksHeading" in node ? node.sulRelLinksHeading : undefined
+  const relLinks = "sulRelLinks" in node ? node.sulRelLinks : undefined
+
+  useEffect(() => {
+    const wysiwygContainers: NodeListOf<Element> = document.querySelectorAll(".wysiwyg")
+
+    const allHeadings: Heading[] = []
+
+    wysiwygContainers.forEach((container, containerIndex) => {
```

Comment:
Don't we want more than just the wysiwyg headings? Cards, and all that should be included I believe.

Reply:
Great question! The figma only seems to include the wysiwyg headings but I'll ask Darcy during out FE + Designer check-in on Monday 🤔